cabal-install/src/Distribution/Client/Init/Interactive/Command.hs

```diff
@@ -357,12 +357,16 @@ licensePrompt flags = getLicense flags $ do
 authorPrompt :: Interactive m => InitFlags -> m String
 authorPrompt flags = getAuthor flags $ do
     name <- guessAuthorName
-    promptStr "Author name" (DefaultPrompt name)
```

Comment:
Do you think it would be worth it to abstract the `guess` mechanism in one helper function? Maybe it could be useful for future additions.

Reply:
Yeah, these two `case`s below look like something one would want to abstract away.